text/0821-public-types.md

```diff
@@ -392,6 +393,28 @@ function takesRouteInfoWithAttributes(routeInfoWithAttributes) {
 ```
 
 
+### `Resolver`
+
+The resolver is a contract implemented by libraries outside Ember itself, such as `ember-resolver`, `ember-strict-resolver`, and any number of custom resolvers which exist in apps across the ecosystem. It has never had public documentation, but is fully public API. It is a user-constructible interface with the following definition (using the `Factory` and `FullName` types exported from the new `@ember/owner` module), available as a new named type-only export from the `@ember/engine` package:
```

Comment:
Is the thinking that since engines are the base entity that accept a resolver, that's the package the type should come from? My kneejerk reaction is that the concept feels more closely tied to `@ember/owner`, but my mental model may be off.

Reply:
That was my own kneejerk reaction as well. I picked `@ember/engine` only because it’s the place that it’s associated with and I usually group “support” modules with their types. I could see an argument for it coming from `@ember/owner` or even `@ember/owner/resolver`, too—I don’t have a strong preference here and so just picked a reasonable-ish place as a starting point for exactly this discussion.

Reply:
In terms of the public API, it is the engine (or application) that's getting constructed with a `Resolver`, so I definitely see where you're coming from. I think what's throwing me off is the knowledge that, under the covers, the key purpose of the resolver is to power the `Owner` (which, concretely, is an engine/application instance; but abstractly we've decided that's its own concept in `@ember/owner`).

Ultimately I don't feel strongly about it, just wanted to make sure I understood the reasoning

Reply:
I brought this up with folks at Framework today, and they agreed with our intuition. 👍🏼